src/main/java/org/opentripplanner/api/parameter/QualifiedModeSet.java

```diff
@@ -29,32 +33,103 @@ public QualifiedModeSet(String s) {
         }
     }
 
-    /**
-     * Modify an existing routing request, setting fields to reflect these qualified modes.
-     * This is intended as a temporary solution, and uses the current system of a single mode set,
-     * accompanied by some flags to help with routing.
-     */
-    public void applyToRoutingRequest(RoutingRequest req) {
+    public RequestModes getRequestModes() {
+        RequestModes requestModes = new RequestModes(
+            StreetMode.WALK,
+            StreetMode.WALK,
+            StreetMode.WALK,
+            Collections.emptySet()
+        );
 
-        if (qModes.isEmpty()) return;
+        if (qModes.isEmpty()) return requestModes;
```

Comment:
Shouldn't this have the default values for transit modes added, now there is no transit modes by default.

Reply:
I haven't looked into how the REST api works in detail, it seems sensible to have it work that way. I am changing it to have a check in the RoutingResource to only set the modes if qModes is not empty. Otherwise it will use the modes from the default RoutingRequest.